packages/app/src/widgets/position-section/position-section.tsx

```diff
@@ -6,24 +6,36 @@ import { useGetBalance, SorobanTokenRecord } from '@/entities/token/hooks/use-ge
 import { WalletContext } from '@/entities/wallet/context/context'
 import { PositionContext } from '@/entities/position/context/context'
 import { useContextSelector } from 'use-context-selector'
-import { PositionCell } from '@/entities/position/types'
+import { PositionCell as PositionCellType } from '@/entities/position/types'
 import { useBorrowDecrease } from '@/features/borrow-flow/hooks/use-borrow-decrease'
 import { useBorrowIncrease } from '@/features/borrow-flow/hooks/use-borrow-increase'
 import { useLendDecrease } from '@/features/borrow-flow/hooks/use-lend-decrease'
-import { useGetCryptocurrencyUsdRates } from '@/entities/currency-rates/hooks/use-get-cryptocurrency-usd-rates'
+import {
+  useGetCryptocurrencyUsdRates,
+  SupportedTokenRates,
+} from '@/entities/currency-rates/hooks/use-get-cryptocurrency-usd-rates'
 import { useLendIncrease } from '@/features/borrow-flow/hooks/use-lend-increase'
 
-const sorobanTokenRecordToPositionCell = (tokenRecord: SorobanTokenRecord): PositionCell => ({
-  value: Number(tokenRecord.balance) / 10 ** tokenRecord.decimals,
-  token: tokenRecord.symbol.substring(1).toLowerCase() as PositionCell['token'],
-})
+const sorobanTokenRecordToPositionCell = (
+  tokenRecord: SorobanTokenRecord,
+  cryptocurrencyUsdRates?: SupportedTokenRates,
+): PositionCellType => {
+  const value = Number(tokenRecord.balance) / 10 ** tokenRecord.decimals
+  const token = tokenRecord.symbol.substring(1).toLowerCase() as PositionCellType['token']
```

Comment:
We can't be sure in that, unfortunately. It's better add a prop to the component with a token code. Or we can extend the type of `tokenRecord` by adding code aside of `SorobanTokenRecord`. Or we can to extend `SorobanTokenRecord` type itself.

Reply:
moved into position provider and refactored

Reply:
thank you